python/cuml/test/experimental/test_permutation_shap.py

```diff
@@ -0,0 +1,166 @@
+from sklearn.datasets import make_regression
+from sklearn.model_selection import train_test_split
+
+
+models_config = ClassEnumerator(module=cuml)
+models = models_config.get_models()
+
+
+# todo: use tags to generate the correct dataset
+@pytest.fixture(scope="session")
+def single_dataset():
+    X, y = make_classification(100, 5, random_state=42)
+    X = X.astype(np.float32)
+    y = y.astype(np.float32)
+    return X, y
+
+
+def func_positional_arg(func):
+    if hasattr(func, "__code__"):
+        all_args = func.__code__.co_argcount
+        if func.__defaults__ is not None:
+            kwargs = len(func.__defaults__)
+        else:
+            kwargs = 0
+        return all_args - kwargs
+    return 2
+
+
+@pytest.mark.parametrize("dtype", [np.float32, np.float64])
+@pytest.mark.parametrize("nfeatures", [4, 100])
+@pytest.mark.parametrize("nbackground", [10, 80])
+@pytest.mark.parametrize("model", [cuml.LinearRegression,
+                                   cuml.KNeighborsRegressor,
+                                   cuml.SVR])
+def test_permutation_gpu_cpu_shap(dtype, nfeatures, nbackground, model):
+    if not has_shap():
+        pytest.skip("Need SHAP installed for these tests")
+
+    import shap
+
+    X, y = make_regression(n_samples=81, n_features=10, noise=0.1,
+                           random_state=42)
+
+    X_train, X_test, y_train, y_test = train_test_split(
+        X, y, test_size=1, random_state=42)
+
+    X_train = X_train.astype(np.float32)
+    X_test = X_test.astype(np.float32)
+    y_train = y_train.astype(np.float32)
+    y_test = y_test.astype(np.float32)
+
+    model = model().fit(X_train, y_train)
+
+    explainer = shap.PermutationExplainer(model.predict, X_train)
+    shap_values = explainer(X_test[0].reshape(1, 10))
+
+    cu_explainer = \
+        cuml.experimental.explainer.PermutationSHAP(model=model.predict,
+                                                    masker=X_train,
+                                                    gpu_model=True)
+
+    cu_shap_values = cu_explainer.shap_values(X_test[0])
+    assert array_equal(cu_shap_values, shap_values,
+                       1e-1, with_sign=True)
+
+
+@pytest.mark.parametrize("model_name", list(models.keys()))
+def test_cuml_models(single_dataset, model_name):
+    n_pos_args_constr = func_positional_arg(models[model_name].__init__)
+
+    if model_name in ["SparseRandomProjection", "GaussianRandomProjection"]:
+        model = models[model_name](n_components=2)
+    elif model_name in ["ARIMA", "AutoARIMA", "ExponentialSmoothing"]:
+        model = models[model_name](np.random.normal(0.0, 1.0, (10,)))
+    else:
+        if n_pos_args_constr == 1:
+            model = models[model_name]()
+        elif n_pos_args_constr == 2:
+            model = models[model_name](5)
+        else:
+            model = models[model_name](5, 5)
+
+    X, y = single_dataset
+
+    X_train, X_test, y_train, y_test = train_test_split(
```

Comment:
life might be nicer if we fixturized the whole combo of: gen dataset, split train/test, ensure fp32